pkg/controller/machineautoscaler/machinetarget.go

```diff
@@ -280,3 +291,67 @@ func (mt *MachineTarget) WarningForInvalidGPUAcceleratorLabel() string {
 
 	return warning
 }
+
+// UpdateScaleFromZeroAnnotations inspects the target's annotations
+// and checks whether scale from zero annotations are present and appends
+// the set of annotations that are missing. We are interested in appending
+// annotations with `capacity.cluster-autoscaler.kubernetes.io` prefix.
+func (mt *MachineTarget) UpdateScaleFromZeroAnnotations() error {
+	annotations := mt.GetAnnotations()
+
+	if annotations == nil {
+		return nil
+	}
+
+	err := checkScaleFromZeroAnnotations(annotations)
+	if err != nil {
+		// canScale, err = checkNewScaleFromZeroAnnotations(annotations)
+		// if !canScale && err != nil {
+		return fmt.Errorf("failed to check scale from zero annotations: %w", err)
+		// }
+	}
+
+	mt.SetAnnotations(annotations)
+
+	return nil
+}
+
+func checkScaleFromZeroAnnotations(annotations map[string]string) error {
+	cpu := parseKey(annotations, cpuKeyOriginal)
+	if cpu != "" {
+		annotations[cpuKey] = cpu
+	}
+
+	// Original memory key specifies the memory in memibytes, thus
+	// we need to convert this intereger into bytes.
+	mem := parseKey(annotations, memoryKeyOriginal)
+	if mem != "" {
+		memInt, err := strconv.ParseInt(mem, 10, 0)
+		if err != nil {
+			return fmt.Errorf("could not parse integer: %w", err)
+		}
+		annotations[memoryKey] = resource.NewQuantity(memInt*util.MiB, resource.DecimalSI).String()
+	}
+
+	// Original GPU annotation is split into 2 upstream annotations
+	gpu := parseKey(annotations, gpuCountKeyOriginal)
+	if gpu != "" {
+		annotations[gpuCountKey] = gpu
+		annotations[gpuTypeKey] = "nvidia.com/gpu"
```

Comment:
could you add a `// TODO ` comment here saying that we need to come back and fix this once we have the proper gpu types.

Reply:
Added.